Add support for input range

robertalexa opened the pull request:
This PR adds support for the input range as a form field.

Screenshot:
![image](https://user-images.githubusercontent.com/9019306/188149730-f2c4cc35-a36f-461e-b700-091605df73f8.png)

By default, input ranges do not have a way to display their current value, so I have added an extra element that follows the handle.

Here is the PR on the docs repo that should ideally be merged in at the same time: https://github.com/wintercms/docs/pull/91

robertalexa committed:
Add support for input range

LukeTowers commented on modules/backend/widgets/form/partials/_field_range.php:
```diff
@@ -0,0 +1,33 @@
+            });
+            input.dispatchEvent(new Event('input'));
+        });
+    </script>
```
Does this work if the field is dynamically displayed after being loaded over AJAX? Typically field JS should be added as it's own JS file (ideally written as a Snowboard plugin now) that binds to events at the document root and filters based on the current input, generally with a selector along the lines of `[data-control="range"]`

LukeTowers commented on modules/backend/widgets/form/partials/_field_range.php:
```diff
@@ -0,0 +1,33 @@
+    />
+    <span style="position: absolute; transform: translateX(-50%)"><?= isset($field->value) ? $field->value : 50 ?></span>
+    <script>
+        $(document).ready(function () {
+            const input = document.getElementById("<?= $field->getId() ?>");
+            input.addEventListener("input", function () {
+                this.nextElementSibling.innerHTML = this.value;
+                var pos = ((this.value - <?= $min ?>) / (<?= $max ?> - <?= $min ?>) * 100);
+                this.nextElementSibling.style.left = `calc(${pos}% + ${8 - pos * 0.15}px)`;
+            });
+            input.dispatchEvent(new Event('input'));
+        });
```
```suggestion
        (() => {
            const input = document.getElementById("<?= $field->getId() ?>");
            input.addEventListener("input", function () {
                this.nextElementSibling.innerHTML = this.value;
                var pos = ((this.value - <?= $min ?>) / (<?= $max ?> - <?= $min ?>) * 100);
                this.nextElementSibling.style.left = `calc(${pos}% + ${8 - pos * 0.15}px)`;
            });
            input.dispatchEvent(new Event('input'));
        })();
```

LukeTowers committed:
Update modules/backend/widgets/form/partials/_field_range.php

LukeTowers committed:
Code review

Handle min being greater than max & set the correct default midpoint value based on the min & max values

bennothommo committed:
Fix code smell (indentation)
